synapse/config/server.py

```diff
@@ -582,7 +611,7 @@ def default_config(self, server_name, data_dir_path, **kwargs):
         # Defaults to 'true'.
         #
         #allow_per_room_profiles: false
-        """ % locals()
+        """ % merge_dicts(locals(), globals())
```

Comment:
I'm not a fan of this tbh, as `globals()` can get quite cluttered, and it could easily clobber locals. Can we not just set `default_room_version = DEFAULT_ROOM_VERSION` or whatever? Or even just have `#default_room_version: "<some_room_version>"`?

Reply:
`merge_dicts` returns a new dict rather than appending to `locals()`.

> Can we not just set default_room_version = DEFAULT_ROOM_VERSION or whatever?

Yeah we can just do this I suppose.

Reply:
> `merge_dicts` returns a new dict rather than appending to `locals()`.

Sure, but the result would produce the value from `globals()` rather than `locals()`, which feels surprising. In general I think this has crossed the threshold of "too magic magic".